File: tests/rosettacode/luhn.py
# Luhn test
def luhn(s):
    digits = [int(c) for c in reversed(s)]
    total = 0
    for i, d in enumerate(digits):
        if i % 2 == 1:
            d *= 2
            if d > 9:
                d -= 9
        total += d
    return total % 10 == 0

for t in ["49927398716", "49927398717", "1234567812345678"]:
    print(f"{t}: {luhn(t)}")

File: tests/rosettacode/luhn.clj
;; Luhn test
(defn luhn? [s]
  (let [digits (map #(- (int %) (int \0)) (reverse s))
        doubled (map-indexed (fn [i d]
                               (if (odd? i)
                                 (let [d2 (* 2 d)]
                                   (if (> d2 9) (- d2 9) d2))
                                 d))
                             digits)]
    (zero? (mod (reduce + doubled) 10))))

(doseq [t ["49927398716" "49927398717" "1234567812345678"]]
  (println (str t ": " (luhn? t))))
